talk/basicconcepts/references.tex

```diff
@@ -29,24 +29,23 @@
   \frametitlecpp[98]{Pointers vs References}
   \begin{block}{Specificities of reference}
     \begin{itemize}
-    \item natural syntax
-    \item must be assigned when defined, cannot be \mintinline{cpp}{nullptr}
-    \item cannot be reassigned
-    \item non-const references to temporary objects are not allowed
+    \item Natural syntax
+    \item Cannot be \mintinline{cpp}{nullptr}
+    \item Must be assigned when defined, cannot be reassigned
+    \item References to temporary objects must be \mintinline{cpp}{const}
     \end{itemize}
   \end{block}
   \begin{block}{Advantages of pointers}
     \begin{itemize}
-    \item can be reassigned to point elsewhere or to \mintinline{cpp}{nullptr}
-    \item clearly indicates that argument may be modified
+    \item Can be \mintinline{cpp}{nullptr}
+    \item Can be initialized after declaration, can be reassigned
     \end{itemize}
   \end{block}
   \pause
   \begin{goodpractice}{References}
     \begin{itemize}
-      \item Always use references when you can
-      \item Consider that a referenced object can be modified
-      \item Use \mintinline{cpp}{const} when it's not the case
+      \item Prefer using references instead of pointers
+      \item Mark references \mintinline{cpp}{const} to prevent modification
```

Comment:
Since you removed the "Consider that a referenced object can be modified": Should we still add something along the lines of: "Remember that non-const references/pointers can modify the referred object" (e.g. in the context of a function you pass that reference/pointer to) or is that just too obvious?

Reply:
Well, the last item tells people to mark as `const` what they don't want to be modified, which implies that if you don't mark them as `const`, they *can* be modified. Maybe this is not too obvious to beginners. I can add back that point if you want, I removed because the slide was too full with 3 items and I thought it was a bit redundant.

Reply:
ok. no further change necessary!